axum/src/routing/path_router.rs

```diff
@@ -111,8 +111,7 @@ where
     }
 
     fn set_node(&mut self, path: &str, id: RouteId) -> Result<(), String> {
-        let mut node =
-            Arc::try_unwrap(Arc::clone(&self.node)).unwrap_or_else(|node| (*node).clone());
+        let mut node = (*self.node).clone();
```

Comment:
Could you still use fully qualified syntax here? Just so it's clear that this is a 'shallow' clone. I suspect we're not consistent about that overall, but would probably use fully qualified if we were to make things consistent.

Reply:
This was actually a deep clone.

However, I took a minute to think about how to actually do what the code probably wanted to do in the first place (which I guess I should have done at the start) and changed it to use `Arc::make_mut`. So as long as the router hasn't been cloned, it'll cheaply update it and otherwise fall back to a clone of `Node`.